src/workloads/query/MatchWithLargeExpression.yml

```diff
@@ -0,0 +1,150 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload tests the performance of expression search for parameter re-use during parameterization. 
+  Before September 2023, parameter re-use had O(n^2) complexity due to using a vector for looking up equivalent expressions. 
+  SERVER-79092 fixed this issue by switching over to a map once the amount of expressions reaches a threshold (currently 50).
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  MaxPhases: &MaxPhases 7
+  DocumentCount: &DocumentCount 10
+
+Keywords:
+- Loader
+- CrudActor
+- QuiesceActor
+- insert
+- Aggregation
+
+ActorTemplates:
+- TemplateName: MatchExpressionWithOrClausesTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "unused"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: {^Parameter: {Name: "Repeat", Default: 1}}
+          Collection: *Collection
+          Operations:
+          - OperationName: aggregate
+            OperationCommand:
+              Pipeline: [
+                {$match: {$or: {^Array: {of: 
+                  {$and: [
+                    {a: {$gt: {^Inc: {start: 10}}}},
+                    {b: {$gt: {^Inc: {start: 10}}}}
+                  ]},
+                  number: {^Parameter: {Name: "NumberOfClauses", Default: -1}}
+                }}}}
+              ]
+
+Actors:
+# Clear any pre-existing collection state.
+- Name: ClearCollection
```

Comment:
You shouldn't really need this actor (although I can see how it makes life easier when running the workload locally).

Maybe a default false ```dropFirst``` attribute on the loader actor would be a reasonable feature request to replace this functionality?

Reply:
created https://jira.mongodb.org/browse/EVG-20779

Reply:
Thank you!